### DISCUSSION TITLE
The docs say that the "display" function is built-in, but built-in to what?

### DISCUSSION THREAD

**Discussion by ttmc** [observablehq/framework#907] [Q&A] [answered]
In the Observable Framework docs (with source at https://github.com/observablehq/framework/blob/main/docs/javascript/display.md ), it says, "The built-in `display` function displays the specified value."

I don't know where the `display` function is coming from. It's not part of standard JavaScript AFAIK, so it's not "built-in to standard JavaScript". I gather that Observable Framework imports a bunch of libraries by default, so I wonder where the `display` function is coming from.

**Comment by mbostock** [comment, marked as answer]
It’s built-in to Observable Framework. I’m not sure if this answers your question, but it’s implemented as a “shadow” variable using the Observable Runtime; each code block has its own `display` function that is uniquely bound to its output location. You can find the implementation here:

https://github.com/observablehq/framework/blob/c114f882fc4c501c678a123058c7f01f6dac1253/src/client/main.js#L52-L55
